Change intermediate column syntax. (#1630)
powdr-labs/powdr: book/src/pil/declarations.md

where the `TV` are newly created type variables that can be used in the type.

This syntax can be used for constants, fixed columns, witness columns and even (higher-order)

> functions that can transform expressions. The kind of symbol is deduced by its type and if
> it is has a value or not:

Reviewer: ```suggestion
functions that can transform expressions. The kind of symbol is deduced by its type and whether
it is has a value:
```
